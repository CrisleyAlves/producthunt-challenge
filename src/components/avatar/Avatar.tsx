import React from "react";

/**
 * Renders a user avatar image with customizable alt text, ARIA description, and additional CSS classes.
 *
 * @param avatarUrl - The URL of the avatar image to display.
 * @param altText - The alternative text for the image, used for accessibility.
 * @param ariaDescription - A description of the image for assistive technologies.
 * @param classNames - Optional additional CSS class names to apply to the image element.
 * @returns The rendered avatar image element.
 */
export default function Avatar({
  avatarUrl,
  altText,
  ariaDescription,
  classNames = "",
}: {
  avatarUrl: string;
  altText: string;
  ariaDescription: string;
  classNames?: string;
}) {
  return (
    <img
      className={`avatar ${classNames}`}
      src={avatarUrl}
      alt={altText}
      aria-description={ariaDescription}
    />
  );
}
