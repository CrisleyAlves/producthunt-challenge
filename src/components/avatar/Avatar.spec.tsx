import React from "react";
import { render } from "@testing-library/react";
import { expect } from "vitest";
import Avatar from "./Avatar";

test("renders the avatar image with correct src and alt", () => {
  const avatarUrl = "https://example.com/avatar.jpg";
  const altText = "User Avatar";
  const ariaDescription = "This is a user avatar";

  const { container } = render(
    <Avatar
      avatarUrl={avatarUrl}
      altText={altText}
      ariaDescription={ariaDescription}
    />
  );

  expect(container).toMatchSnapshot();
});
