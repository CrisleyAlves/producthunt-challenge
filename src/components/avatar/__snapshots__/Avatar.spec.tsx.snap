// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders the avatar image with correct src and alt 1`] = `
<div>
  <img
    alt="User Avatar"
    aria-description="This is a user avatar"
    class="avatar "
    src="https://example.com/avatar.jpg"
  />
</div>
`;
